test/create_test_bag.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# Create a rosbag with the supported types
+import rosbag
+from ros_numpy import msgify, numpy_msg
+import numpy as np
+from sensor_msgs.msg import Image, PointCloud2, CompressedImage
+from nav_msgs.msg import OccupancyGrid
+from geometry_msgs.msg import Vector3, Point, Quaternion, Transform, Pose
+import zlib, struct
+
+
+def makeArray(npoints):
+	points_arr = np.zeros((npoints,), dtype=[
+		('x', np.float32),
+		('y', np.float32),
+		('z', np.float32),
+		('r', np.uint8),
+		('g', np.uint8),
+		('b', np.uint8)])
+	points_arr['x'] = np.random.random((npoints,))
+	points_arr['y'] = np.random.random((npoints,))
+	points_arr['z'] = np.random.random((npoints,))
+	points_arr['r'] = np.floor(np.random.random((npoints,)) * 255)
+	points_arr['g'] = 0
+	points_arr['b'] = 255
+
+	return points_arr
+
+# From http://stackoverflow.com/questions/902761/saving-a-numpy-array-as-an-image
+def write_png(buf, width, height):
```

Comment:
I'd prefer not to build a mini PNG library into this package! How about:
```python
from io import BytesIO
import imageio

def write_png(arr):
    b = BytesIO()
    imageio.imwrite(b, arr, format='png')
    return b.getvalue()
```

Reply:
I prefer to not pull another library as dependency, even being a test-only dependency.

It's just a test and it works, sorry to not agree.

Reply:
Why is writing a PNG even a necessary part of this test? That seems pretty out of scope for checking the mangling of type ids by rosbag, which is what this PR is about, right?